### DISCUSSION TITLE
How to manage node state?

### DISCUSSION THREAD

**Discussion by neerajshdev** [bumble-tech/appyx#714] [General]
How to persit state accrod config changes and process death?        
I knows 
```
nodeContext.getRetainedInstance("key") {  }
```
But this does not persit state across process death.

**Comment by KovalevAndrey** [comment]
hey @Neerajsh8851, you need to save/restore the state properly, you can do it inside Node:
```
class ContainerNode internal constructor(
    buildContext: BuildContext,
    private val backStack: BackStack<NavTarget> = BackStack(
        initialElement = Picker,
        savedStateMap = buildContext.savedStateMap,
    )
) : ParentNode<NavTarget>(
    navModel = backStack,
    buildContext = buildContext,
) {
    
    init {
        val restoredValue = buildContext.savedStateMap?.getValue("key") as? String 
    }

    override fun onSaveInstanceState(state: MutableSavedStateMap) {
        super.onSaveInstanceState(state)
        state["key"] = "value"
    }
```

or implement plugin and pass it to list of plugins of the Node

<img width="716" alt="0981754e-6017-4b4d-93ee-c7a4ea0f0ad3" src="https://github.com/user-attachments/assets/44a60d76-fda2-4006-b8ae-a46a07e2ba54">

**Reply by neerajshdev** [reply]
Thanks !
But the onSavedInstaceState is final in version 2.x I cannot override it.

**Reply by KovalevAndrey** [reply]
@Neerajsh8851 you're right, it should be protected as it is in 1.x. Can you use `SavesInstanceState` plugin until it's fixed ?

**Reply by neerajshdev** [reply]
Yes
